src/audio/SDL_audiotypecvt.c

```diff
@@ -578,10 +578,10 @@ static void SDLCALL SDL_Convert_F32_to_S8_SSE2(SDL_AudioCVT *cvt, SDL_AudioForma
         }
     }
 
-    SDL_assert(!i || ((((size_t)dst) & 15) == 0));
+    SDL_assert(!i || (!(((size_t)dst) & 15)));
```

Comment:
Extra parens around `(!(((size_t)dst) & 15))` can be removed so the line can read
`    SDL_assert(!i || !(((size_t)dst) & 15));`